### PR TITLE
Implement core go-sdk

### PR DESCRIPTION
**Pull request by denalimarsh**
The Kava Go SDK provides unique types and functionality required by services that interact with Kava's core modules. It includes the following core components:

client - types required by bep3-deputy's rpc client and initialization of local client.
tx - base types for standard txs and msgs, tx parsing and result types for bep3-deputy's rpc client, codec registration.
keys - management of private keys and account recovery from mnenomic phrase.
types - types required by bep3-deputy's executor.

This PR contains the minimum types and features such that a Kava specific bep3-deputy Executor can be implemented.

### CODE REVIEW

File: types/swap.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+func (bz SwapBytes) Marshal() ([]byte, error) {
+	return bz, nil
+}
+
+func (bz *SwapBytes) Unmarshal(data []byte) error {
+	*bz = data
+	return nil
+}
+
+func (bz SwapBytes) MarshalJSON() ([]byte, error) {
```
Comment (karzak) [inline types/swap.go]: Can we use the built in Marshal/UmarshalJSON methods of the amino codec?

File: types/fixed8.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	if err := json.Unmarshal(data, &fl); err != nil {
+		return err
+	}
+
+	*f = Fixed8(float64(Fixed8Decimals) * fl)
+	return nil
+}
+
+// MarshalJSON implements the json marshaller interface
+func (f *Fixed8) MarshalJSON() ([]byte, error) {
```
Comment (karzak) [inline types/fixed8.go]: Same question as `swap.go`. Fine if we need custom methods, but reduces size of the code base if we can assume that we have access to an amino codec and can marshal/unmarshal using those methods.

File: keys/keys_test.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	TestMnenomic     = "equip town gesture square tomorrow volume nephew minute witness beef rich gadget actress egg sing secret pole winter alarm law today check violin uncover"
+	TestExpectedAddr = "kava1gflyk57p2ppflqjszyak29x62wuw955eqrzcmy"
```
Comment (karzak) [inline keys/keys_test.go]: When i manually test that mneomic using `kvcli keys add dry --dry-run --recover` I get kava15qdefkmwswysgg4qxgqpqr35k3m49pkx2jdfnw

File: keys/keys.go [hunk 4]
```diff
@@ -12,28 +12,32 @@ import (
...
 	"github.com/tendermint/tendermint/crypto/secp256k1"
 )
 
 const (
 	defaultBIP39Passphrase = ""
-	// TODO: Change BIP44Prefix
-	BIP44Prefix = "44'/714'/"
-	PartialPath = "0'/0/0"
-	FullPath    = BIP44Prefix + PartialPath
+	BIP44Prefix            = "44'/188'/"
```
Comment (karzak) [inline keys/keys.go]: Should be `118` not `188` (explains test output)

File: keys/keys_test.go (hunk 3 above)
Comment (karzak) [reply keys/keys_test.go]: see below, there's a typo in the BIP44Prefix

File: types/fixed8.go (hunk 2 above)
Comment (denalimarsh) [reply types/fixed8.go]: https://github.com/Kava-Labs/go-sdk/pull/1/commits/b500f5d8773b86f005d2d6ea4ec750013e370060

File: types/swap.go (hunk 1 above)
Comment (denalimarsh) [reply types/swap.go]: https://github.com/Kava-Labs/go-sdk/pull/1/commits/b500f5d8773b86f005d2d6ea4ec750013e370060

File: types/account.go [hunk 5]
```diff
@@ -0,0 +1,186 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pkg/errors"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+// AppAccount definition
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply types/account.go]: Will do as part of Asana task - "_go-sdk PR: polish up go-sdk (add README.md + unit tests) and prepare it for future expansion_"

File: keys/keys.go (hunk 4 above)
Comment (denalimarsh) [reply keys/keys.go]: https://github.com/Kava-Labs/go-sdk/pull/1/commits/64d1867e55f9a57b0d3c1c7b21d937ee0b874d62

File: keys/keys_test.go (hunk 3 above)
Comment (denalimarsh) [reply keys/keys_test.go]: https://github.com/Kava-Labs/go-sdk/pull/1/commits/64d1867e55f9a57b0d3c1c7b21d937ee0b874d62

File: types/fixed8.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
+package types
```
[the remark this answers is not in the document]
Comment (denalimarsh) [reply types/fixed8.go]: https://github.com/Kava-Labs/go-sdk/pull/1/commits/1b60045cfa111f0992a73ad365b4670af62c8660

**Pull request merged**